### PR TITLE
New with parentheses

### PR DESCRIPTION
**Pull request by carusogabriel**
Something to start with: https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.10/src/Fixer/Operator/NewWithBracesFixer.php

Already proposed and voted in https://github.com/doctrine/coding-standard/pull/9#issuecomment-353872306

### CODE REVIEW

**Comment by kukulich** [comment]
@carusogabriel I can write the sniff for https://github.com/slevomat/coding-standard, would it be ok?

**Comment by carusogabriel** [comment]
@kukulich Perfect. I'm not good at sniffers, so I appreciate your help :smile:

**Comment by kukulich** [comment]
Please test: https://github.com/slevomat/coding-standard/commit/a3c36e949745424cdf90052065bc17d739fd4d75

File: tests/fixed/new-with-parentheses.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+$foo = new \DateTimeImmutable();
+
+$barClassName = 'Bar';
+$bar 		  = new $barClassName();
```
Comment (lcobucci) [inline tests/fixed/new-with-parentheses.php]: This `=` should be aligned with the one in the line above (this file only)

File: tests/fixed/new-with-parentheses.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+$classNamesInArray = ['Baz'];
+$foo 			   = new $classNamesInArray[0]();
```
Comment (lcobucci) [inline tests/fixed/new-with-parentheses.php]: This `=` should be aligned with the one in the line above (this file only)

File: tests/fixed/new-with-parentheses.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+$classNamesInObject 	 = new stdClass();
```
Comment (lcobucci) [inline tests/fixed/new-with-parentheses.php]: This `=` should be aligned with the one in the line bellow (this file only)

File: tests/fixed/new-with-parentheses.php [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+$classNamesInObject->foo = 'Foo';
+$foo 					 = new $classNamesInObject->foo();
```
Comment (lcobucci) [inline tests/fixed/new-with-parentheses.php]: This `=` should be aligned with the one in the line above (this file only)

File: lib/Doctrine/ruleset.xml [hunk 5]
```diff
@@ -95,6 +95,8 @@
     <rule ref="SlevomatCodingStandard.ControlStructures.DisallowEqualOperators"/>
     <!-- Require language constructs without parentheses -->
     <rule ref="SlevomatCodingStandard.ControlStructures.LanguageConstructWithParentheses"/>
+    <!-- Require new instances with parentheses -->
+    <rule ref="SlevomatCodingStandard.ControlStructures.NewWithParenthesesSniff"/>
```
Comment (carusogabriel) [inline lib/Doctrine/ruleset.xml]: @kukulich What is the correct Sniff name? Travis isn't find it :cry:

Comment (kukulich) [reply lib/Doctrine/ruleset.xml]: @carusogabriel Remove "Sniff" at the end :)

```
<rule ref="SlevomatCodingStandard.ControlStructures.NewWithParentheses"/>
```

**Comment by carusogabriel** [comment]
@Ocramius @Majkl578 @lcobucci How should we proceed now?

File: composer.json [hunk 6]
```diff
@@ -18,7 +18,7 @@
         "php": "^7.1",
         "squizlabs/php_codesniffer": "^3.2",
         "dealerdirect/phpcodesniffer-composer-installer": "^0.4.2",
-        "slevomat/coding-standard": "^4.3.0"
+        "slevomat/coding-standard": "dev-master"
```
Comment (Ocramius) [inline composer.json]: This needs to become a stable dependency before merging

**Comment by Ocramius** [comment]
@carusogabriel I'd say this is good to go. We just need a stable version of `slevomat/coding-standard` and then a new major release.

**Comment by kukulich** [comment]
@carusogabriel I only wait for confirmation that https://github.com/slevomat/coding-standard/issues/258 is ok and I'll do a release.

File: composer.json (hunk 6 above)
Comment (Majkl578) [reply composer.json]: + requires bump once tagged in Slevomat CS. 👍

**Comment by Majkl578** [comment]
@carusogabriel Version bumped in 979478d.

**Comment by Majkl578** [comment]
> Merge branch 'master' into new-with-parentheses 

Please, always rebase on top of master, not merge it into PR branch.

File: composer.json (hunk 6 above)
Comment (mikeSimonson) [reply composer.json]: 4.4 is out

**Pull request merged**